modules/formulize/include/formdisplay.php

```diff
@@ -395,8 +395,13 @@ function displayForm($formframe, $entry="", $mainform="", $done_dest="", $button
 
 	$mid = getFormulizeModId();
 
-	$currentURL = getCurrentURL();
-
+	$currentURL = ( (isset($_GET['ve']) && isset($_GET['fid'])) ? $_SERVER['PHP_SELF'] . "?fid=" . $_GET['fid'] : getCurrentURL() );

```

Comment:
There's an issue with this approach. If you have a form with two list screens, and you open the non-default list screen and edit an entry, when you click the Save And Leave or the Leave button, instead of returning to the list screen you were viewing, you will now be directed to the other list screen because it is the default.